test/test_broker/iaas.CloudProviderClient.js

```diff
@@ -13,6 +13,191 @@ describe('iaas', function () {
+          VolumeType: volType,
+          TagSpecifications: [{
+            ResourceType: 'volume',
+            Tags: [{
+              Key: 'customkey',
+              Value: 'customvalue'
+            }]
+          }]
+        }).returns({
+          promise: () => Promise.reject(new Error('diskerror'))
+        });
+        return client
+          .createDiskFromSnapshot(snapshotId, zone, {
+            volumeType: 'gp2',
+            tags: {
+              customkey: 'customvalue'
+            }
+          })
+          .catch(err => {
+            expect(err.message).to.eql('diskerror');
+          });
+      });
+      it('should fail on waiting for disk from snapshot for aws', function () {
+        const client = new CloudProviderClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const createVolStub = sinon.stub(client.blockstorage, 'createVolume');
+        createVolStub.withArgs({
+          AvailabilityZone: zone,
+          SnapshotId: snapshotId,
+          VolumeType: volType,
+          TagSpecifications: [{
+            ResourceType: 'volume',
+            Tags: [{
+              Key: 'customkey',
+              Value: 'customvalue'
+            }]
+          }]
+        }).returns({
+          promise: () => Promise.resolve(response)
+        });
+        const waitForStub = sinon.stub(client.blockstorage, 'waitFor');
+        waitForStub.withArgs('volumeAvailable', {
+            VolumeIds: [diskId]
+          })
+          .returns({
+            promise: () => Promise.reject(new Error('diskwaiterror'))
+          });
+        return client
+          .createDiskFromSnapshot(snapshotId, zone, {
+            volumeType: 'gp2',
+            tags: {
+              customkey: 'customvalue'
+            }
+          })
+          .catch(err => {
+            expect(err.message).to.eql('diskwaiterror');
+          });
+      });
+    });
+
+    describe('#getDiskMetadata', function () {
+      const diskId = 'sample-disk';
+      const response = {
+        Volumes: [{
+          VolumeId: diskId,
+          Size: '4',
+          AvailabilityZone: 'zone',
+          VolumeType: 'type'
+        }]
+      };
+      it('should fetch disk metadata for aws', function () {
+        const client = new CloudProviderClient({
+          name: 'aws',
+          key: 'key',
+          keyId: 'keyId',
+          region: 'region'
+        });
+        const describeVolumesStub = sinon.stub(client.blockstorage, 'describeVolumes');
+        describeVolumesStub.withArgs({
+          VolumeIds: [diskId]
+        }).returns({
+          promise: () => Promise.resolve(response)
+        });
+        return client
+          .getDiskMetadata(diskId)
+          .then(disk => {
+            expect(disk.volumeId).to.eql(diskId);
+            expect(disk.size).to.eql('4');
+            expect(disk.zone).to.eql('zone');
+            expect(disk.type).to.eql('type');
```

Comment:
Same as above, please verify all keys in response.

Reply:
Done in latest commit